hub/util/exceptions.py

```diff
@@ -4,3 +4,34 @@ def __init__(
         message="If the size of the last chunk is given, it must be smaller than the requested chunk size.",
     ):
         super().__init__(message)
+
+
+class InvalidBytesRequestedError(Exception):
+    def __init__(self):
+        super().__init__(
+            "The byte range provided is invalid. Ensure that start_byte <= end_byte and start_byte > 0 and end_byte > 0"
+        )
+
+
+class S3GetError(Exception):
+    """Catchall for all errors encountered while working getting an object from S3"""
```

Comment:
That's easier said than done, there could be a lot of things that go wrong on AWS. I've included a todo for this, I think this task will take some time, will need to explore all the possibilities.
This could be a good issue for someone to get started on Hub 2.0. I'll create an issue for this once this PR gets merged.

Reply:
Good call. Maybe we can set up a few test scenarios (bad internet connection, corrupted bucket, etc) and see what errors come back. Like QA but intentionally breaking it for more insight into what issues translate to which errors.

AWS might also have a list of things that can cause exceptions on certain calls, it might be good to link to something like that at least for now